config/environments/development.rb

```diff
@@ -5,13 +5,13 @@
     ENV[key.to_s] = value
   end if File.exists?(env_file)
 
-
   ENV['DISTRICT_KEY'] = 'somerville'
   ENV['DISTRICT_NAME'] = 'Localhost Public Schools'
   ENV['USE_MOCK_LDAP'] = 'true'
   ENV['MOCK_LDAP_PASSWORD'] = 'demo-password'
   ENV['ENABLE_CLASS_LISTS'] = 'true'
   ENV['HOUSEMASTERS_AUTHORIZED_FOR_GRADE_8'] = 'true'
+  ENV['AWS_S3_PHOTOS_BUCKET'] = 'fake-demo-student-insights-photos'
```

Comment:
So this is real, public and doesn't require credentials to read or write?  That's probably worth a comment here.  Another improvement we could make some other time is to make a fake S3 client object (like with LDAP), and have the `put_object` and `get_object` and methods just write into `tmp/photos` or something on the filesystem so there's no external things involved.

Reply:
Hey @kevinrobinson! Good thoughts. 

+ The bucket is real. Credentials are required to: list objects, write objects, read bucket permissions and write bucket permissions. 
+ The image is readable by everyone. Credentials are required to write/update the image. 
+ A code comment here is a good idea! Thanks for suggesting.
+ What are some of the pros/cons of mocking S3 instead of calling a real S3 bucket in development? 
  + For LDAP, setting up a real LDAP service is non-trivial and would involve learning a new technology, so mocking LDAP was much quicker to implement. 
  + One pro to using the real service is that it's better for development — if there's a mistake in the way the code calls S3, using a real bucket for development is more likely to surface that mistake than a mock. We depend on ourselves to build an accurate mock object that handles edge cases correctly. 
  + One con to using a real service is that it introduces the possibility of confusion. i named the bucket `"fake-demo-student-insights-photos"` to make its function as explicit as possible.
    + If somehow we screw up big-time and the demo bucket is used in production, there will be no effect because none of the student photo records in the database will match S3.
    + If somehow we screw up big-time and the production bucket is used on the demo Heroku site, we have additional layers of protection because the S3 creds need to be set in Heroku to access those images.
    + If somehow we screw up big-time and send real photos to the demo bucket, they'll be private, non-readable/writeable to others by default and we can quickly delete them from S3.
+ So my overall thought is that using this S3 bucket in development/demo Heroku site is very safe. But I may be missing something and I'm curious to hear your thoughts on this!

Reply:
@alexsoble This is awesome and super thorough!  I see the same things you're seeing with regards to security, it seems like there are a lot of steps here that would require us to make a series of mistakes to confuse this with a production bucket, which is great!

I was thinking about this more in terms of the developer experience, and isolation across different environments.  I see the service is mocked out in test so that test runs locally and in Travis won't use this bucket if I follow.  So next I'm wondering about if the same bucket would be used in development locally by everyone on the team, or maybe on the demo site too, and whether that involves coordinating and sharing credentials.

Your pros and cons for mocking/using a real service are great too!  I do think that the way we are using the S3 API is simple enough (eg, get/put) where it could be a good candidate for mocking the service in development.